### PR TITLE
Create Ocean Spark cluster within existing VPC

### PR DESCRIPTION
**Pull request by jrj-d**

### CODE REVIEW

File: examples/from-private-vpc/main.tf [hunk 1]
```diff
@@ -0,0 +1,150 @@
+terraform {
+  required_providers {
+    spotinst = {
+      source  = "spotinst/spotinst"
+      version = "~> 1.64"
+    }
+  }
```
Comment (ImpSy) [inline examples/from-private-vpc/main.tf:7]: You should lock all providers (aws and kubernetes are missing here)

File: examples/from-private-vpc/main.tf [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+      ipv6_cidr_blocks = ["::/0"]
+    }
+  }
+
+}
+
+################################################################################
+# Create aws-auth configmap
+# (the eks module recently removed their support for aws-auth management (>=18))
+################################################################################
```
Comment (ImpSy) [inline examples/from-private-vpc/main.tf:60]: Why not use an oldest version of the module like we already do in the other example (and ocean-eks uses internally) ?

Comment (jrj-d) [reply examples/from-private-vpc/main.tf:60]: To keep up with TF. At some point, new version of TF or aws will be incompatible of the version of aws-eks we pinned due to this configmap.
Besides, customers become suspicious when you use old versions of modules. This suspicion is not warranted most of the time, but we must avoid it nonetheless.

File: examples/from-private-vpc/main.tf (hunk 1 above)
Comment (jrj-d) [reply examples/from-private-vpc/main.tf]: Done, thank!

**Review by ImpSy — APPROVED** [review]
LGTM
1 point though: should we split the  backend + providers declaration in a dedicated file to ease reading by separatint init code from "real" logic (I've done it this way in the other examples) ?

**Comment by jrj-d** [comment]
I'll extract version pinning like you did but keep providers within main: providers are filled out with values set by modules (see kubernetes provider) so I like to keep them in the flow for easier reading

**Pull request merged**